woocommerce/woocommerce
# Product Filters > Clear button: Refactor the Product Filters to use a function instead of the hook

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #51636 .

This PR refactors the current useProductFilterClearButtonManager and transforms it into a util function that can be used by each individual filter when the "Show clear button" setting is modified. I added E2E tests in the previous PR to ensure the functionality will continue working as expected.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

> [!NOTE]  
> **Pre-requisite**
> - If you're using WC Beta Tester plugin, make sure the `experimental-blocks` feature is enabled (Tools > WCA Test Helper > Features).

1. Navigate to Product Catalog template (Appearance > Editor > Template > Product Catalog).
2. Add the Product Filters (Experimental) block.
3. For each individual filter, select it and open the Inspector Controls.
4. Disable the "Clear" button and ensure it's hidden in the Editor.
5. Save the template and visit the front page to ensure it's hidden there as well.
6. Go back to the Product Catalog template and enable the "Clear" button;
7. Make sure the Clear button is restored.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment  Refactor individual filters to use a function instead of a hook.

</details>
